modules/wallet.go

```diff
@@ -24,115 +24,97 @@ type WalletInfo struct {
+	// index of the siafund input within the transaction. When 'Sign' is
+	// called, this input will be left unsigned.
+	AddSiafundInput(types.SiafundInput) uint64
+
+	// AddSiafundOutput adds a siafund output to the transaction, returning the
+	// index of the siafund output within the transaction.
+	AddSiafundOutput(types.SiafundOutput) uint64
+
+	// AddArbitraryData adds arbitrary data to the transaction, returning the
+	// index of the data within the transaction.
+	AddArbitraryData(arb []byte) uint64
+
+	// AddTransactionSignature adds a transaction signature to the transaction,
+	// returning the index of the signature within the transaction. The
 	// signature should already be valid, and shouldn't sign any of the inputs
-	// that were added by calling 'FundTransaction'. The updated transaction
-	// and the index of the new signature are returned.
-	AddTransactionSignature(int, types.TransactionSignature) (uint64, error)
-
-	// SignTransaction will sign and delete a transaction, specified by
-	// registration id. If the whole transaction flag is set to true, then the
-	// covered fields object in each of the transaction signatures will have
-	// the whole transaction field set. Otherwise, the flag will not be set but
-	// the signature will cover all known fields in the transaction (see an
-	// implementation for more clarity). After signing, a transaction set will
-	// be returned that contains all parents followed by the transaction. The
-	// transaction is then deleted from the builder registry.
-	SignTransaction(id int, wholeTransaction bool) (txnSet []types.Transaction, err error)
-
-	// ViewTransaction returns a transaction-in-progress along with all of its
-	// parents, specified by id. An error is returned if the id is invalid.
-	// Note that ids become invalid for a transaction after 'SignTransaction'
-	// has been called because the transaction gets deleted.
-	ViewTransaction(id int) (txn types.Transaction, parents []types.Transaction, err error)
+	// that were added by calling 'FundSiacoins' or 'FundSiafunds'.
+	AddTransactionSignature(types.TransactionSignature) uint64
+
+	// Sign will sign any inputs added by 'FundSiacoins' or 'FundSiafunds' and
+	// return a transaction set that contains all parents prepended to the
+	// transaction. If more fields need to be added, a new transaction builder
+	// will need to be created.
+	//
+	// If the whole transaction flag  is set to true, then the whole
+	// transaction flag will be set in the covered fields object. If the whole
+	// transaction flag is set to false, then the covered fields object will
+	// cover all fields that have already been added to the transaction, but
+	// will also leave room for more fields to be added.
+	Sign(wholeTransaction bool) ([]types.Transaction, error)
+
+	// View returns the incomplete transaction along with all of its parents.
+	View() (txn types.Transaction, parents []types.Transaction)
```

Comment:
why does `View` need to return the parents? I'm not totally clear on this, but it doesn't seem like the TransactionBuilder ever modifies the set of parents.